pkg/deserialization/deserializer.go

```diff
@@ -0,0 +1,11 @@
+package deserialization
```

Comment:
Just a thought: 
Could we merge the Serializer & Deserializer interfaces into 1 interface like `SerDe` and have 2 implementations (1 for json & 1 for proto)?
For eg. for json, it could be 

```
JsonSerDe {
     func JSONSerializer() Serializer {
	return SerializeFunc(func(m interface{}) ([]byte, error) {
		return json.Marshal(m)
	})
    }
  
    func JSONDeserializer() Deserializer {
	return DeserializeFunc(func(b []byte, i interface{}) error {
		return json.Unmarshal(b, i)
	})
    }
}
```

I'm suggesting above as it makes the code even more concise and couples the SerDe into 1 interface.
Secondly, we are not going to let clients define custom serdes as well.

Reply:
Actually, I don't agree with this, In Go, it's better to have as small interfaces as possible. We can definitely compose them. [Refer](https://medium.com/@meeusdylan/go-interfaces-keep-them-small-f148ae200d6b)

Reply:
Sure. If these are reusable modules that work outside the scope of Raccoon, then it makes sense to keep them small I guess.
For us: it's too many files to maintain. There's json.go in 2 different packages and these are not usable outside raccoon.

Please feel free to resolve this comment.